lib/keystore/encrypt.go

```diff
@@ -137,8 +137,17 @@ func EncryptAndWriteToFile(file *os.File, pk crypto.PrivateKey, password []byte)
 		return err
 	}
 
-	_, err = file.Write(append(data, byte('\n')))
-	return err
+	_, err = file.Write(data)
+	if err != nil {
+		return err
+	}
+
+	_, err = file.Write([]byte{'\n'})
+	if err != nil {
+		return err
+	}
+
+	return file.Close()
```

Comment:
the `file.Close` needs to be placed into a defer statement since we should close a file whenever an error occurs

Reply:
Totally. In the end I decided to change `EncryptAndWriteToFile` to handle both opening the file and closing it. It didn't make much sense to just do one or the other, and it was also easy to forget to close the file (we forgot in our unit tests). So now it's more of an 'atomic' operation where it does everything in the function. It doesn't matter from the current usage we make of it as well. I also further simplified using `os.WriteFile` to handle the open+write+close for us.